### ISSUE TITLE
Spyder update not checking in conda-forge

### ISSUE THREAD

**Issue by norweeg** [spyder-ide/spyder#21365]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

For Spyder running in a conda environment, Spyder checks Anaconda's channel for updates, regardless of channel configuration set by conda's .condarc file because the channel's URL is hard-coded into Spyder.  Not only will this produce inaccurate results if one is using [Spyder installed from a channel such as conda-forge](https://anaconda.org/conda-forge/spyder), which may have different versions of Spyder available for install than the Anaconda channel, but may also be putting them in violation of the [Anaconda TOS](https://legal.anaconda.com/policies/en/?name=terms-of-service#terms-of-service) for usage of the Anaconda channel if they are using Spyder in a large business (> 200 employees, as defined in the TOS) setting.  The problem is specifically [here](https://github.com/spyder-ide/spyder/blob/a894eb65778777aec915d1d95fe8e564e177cf62/spyder/workers/updates.py#L86).  This should not be hard-coded and, should instead, check for Spyder updates in the users's configured conda channels.

### What steps reproduce the problem?

1. Start monitoring network activity for domains accessed
2. Install Spyder from conda-forge or any non-anaconda conda channel in an environment in which neither "anaconda" nor "defaults" are in the channels key of the applicable .condarc files
3. Open Spyder and check for updates

### What is the expected output? What do you see instead?

Spyder checks for updates in Anaconda channel regardless of channel settings in .condarc file.  It should, instead, check for Spyder updates in the channels configured in the user's [.condarc file(s)](https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html)

## Versions

* Spyder version: 5.4.5
* Python version:  3.11.5
* Qt version: 5.15.8
* PyQt version: 5.15.9
* Operating System name/version: Windows 11

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                                                        :  1.4.1 (OK)
chardet >=2.0.0                                                             :  5.2.0 (OK)
cloudpickle >=0.5.0                                                         :  2.2.1 (OK)
cookiecutter >=1.6.0                                                        :  2.3.0 (OK)
diff_match_patch >=20181111                                                 :  20230430 (OK)
intervaltree >=3.0.2                                                        :  3.1.0 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0,!=8.11.0,!=8.12.0,!=8.12.1 :  8.15.0 (OK)
jedi >=0.17.2,<0.19.0                                                       :  0.18.2 (OK)
jellyfish >=0.7                                                             :  1.0.0 (OK)
jsonschema >=3.2.0                                                          :  4.19.0 (OK)
keyring >=17.0.0                                                            :  24.2.0 (OK)
nbconvert >=4.0                                                             :  7.8.0 (OK)
numpydoc >=0.6.0                                                            :  1.5.0 (OK)
paramiko >=2.4.0                                                            :  3.3.1 (OK)
parso >=0.7.0,<0.9.0                                                        :  0.8.3 (OK)
pexpect >=4.4.0                                                             :  4.8.0 (OK)
pickleshare >=0.4                                                           :  0.7.5 (OK)
psutil >=5.3                                                                :  5.9.5 (OK)
pygments >=2.0                                                              :  2.16.1 (OK)
pylint >=2.5.0,<3.0                                                         :  2.17.5 (OK)
pylint_venv >=3.0.2                                                         :  3.0.2 (OK)
pyls_spyder >=0.4.0                                                         :  0.4.0 (OK)
pylsp >=1.7.4,<1.8.0                                                        :  1.7.4 (OK)
pylsp_black >=1.2.0,<3.0.0                                                  :  1.3.0 (OK)
qdarkstyle >=3.0.2,<3.2.0                                                   :  3.1 (OK)
qstylizer >=0.2.2                                                           :  0.2.2 (OK)
qtawesome >=1.2.1                                                           :  1.2.3 (OK)
qtconsole >=5.4.2,<5.5.0                                                    :  5.4.4 (OK)
qtpy >=2.1.0                                                                :  2.4.0 (OK)
rtree >=0.9.7                                                               :  1.0.1 (OK)
setuptools >=49.6.0                                                         :  68.2.2 (OK)
sphinx >=0.6.6                                                              :  7.2.6 (OK)
spyder_kernels >=2.4.4,<2.5.0                                               :  2.4.4 (OK)
textdistance >=4.2.0                                                        :  4.5.0 (OK)
three_merge >=0.1.1                                                         :  0.1.1 (OK)
watchdog >=0.10.3                                                           :  3.0.0 (OK)
zmq >=22.1.0                                                                :  25.1.1 (OK)

# Optional:
cython >=0.21                                                               :  3.0.2 (OK)
matplotlib >=3.0.0                                                          :  3.8.0 (OK)
numpy >=1.7                                                                 :  1.26.0 (OK)
pandas >=1.1.1                                                              :  2.1.0 (OK)
scipy >=0.17.0                                                              :  1.11.2 (OK)
sympy >=0.7.3                                                               :  1.12 (OK)

# Spyder plugins:
spyder_notebook.notebookplugin 0.5.1                                        :  0.5.1 (OK)

```

**Comment by ccordoba12** [comment]
Hey @norweeg, thanks for bringing this up. I think it's a valid request and something we haven't address so far because our number of downloads from Conda-forge seems rather low compared to defaults.

But we'll fix it in our next version (5.5.0), to be released next month.

**Comment by norweeg** [comment]
> Hey @norweeg, thanks for bringing this up. I think it's a valid request and something we haven't address so far because our number of downloads from Conda-forge seems rather low compared to defaults.
> 
> But we'll fix it in our next version (5.5.0), to be released next month.

Thanks, @ccordoba12!  By the way, you have significantly more (~5x) the installs of spyder from the conda-forge channel than from anaconda channel according to the channel statistics for the spyder package on anaconda.org. (conda-forge has 1852405 where as anaconda only has 373940)

![image](https://github.com/spyder-ide/spyder/assets/5969754/8f845348-e694-41b7-992e-7ef317484711)

**Comment by ccordoba12** [comment]
That's right, but the `anaconda` channel in anaconda.org is not the same as the same as the `defaults` channel. In `defaults` we get ~150k downloads per month last time I checked.
